src/fs_store.rs

```diff
@@ -237,14 +239,14 @@ impl<'a> From<&'a DocumentId> for DocIdPaths {
 
 impl DocIdPaths {
     fn parse<P1: AsRef<Path>, P2: AsRef<Path>>(level1: P1, level2: P2) -> Option<Self> {
```

Comment:
Oh, there is indeed no reason for passing level1 and level2 paths separately. I'll change this.